python/llm/example/GPU/QLoRA-FineTuning/alpaca-qlora/alpaca_qlora_finetuning.py

```diff
@@ -176,9 +178,9 @@ def train(
     else:
         # According to the QLoRA paper, using "nf4" could yield better model quality than "int4"
         # Default 4-bit format for qa-lora is sym_int4
-        if qa_lora:
+        if training_mode == "qalora":
             low_bit_format = "sym_int4"
-        elif lora:
+        elif training_mode == "lora":
             low_bit_format = "bf16"
         else:
             low_bit_format = "nf4"
```

Comment:
check if `training_mode` is valid

Reply:
Have added check in line 115-116 of this file:
```python
invalidInputError(training_mode in ["qlora", "qalora", "lora"],
                  "Only qlora / qalora / lora are supported for training_mode now.")
```